testkit/src/main/java/org/apache/calcite/test/DiffRepository.java

```diff
@@ -895,7 +898,17 @@ private static class Key {
     DiffRepository toRepo() {
       final URL refFile = findFile(clazz, ".xml");
       final String refFilePath = Sources.of(refFile).file().getAbsolutePath();
-      final String logFilePath = refFilePath.replace(".xml", "_actual.xml");
+      final String logFilePath;
+      if (StringUtils.containsIgnoreCase(refFilePath, ".jar!")) {
```

Comment:
I notice this block is identical to the one above, can we put it somewhere helpful and refactor